tests/engine/processes/test_builder.py

```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Tests for `aiida.engine.processes.builder.ProcessBuilder`."""
+import pytest
+
+from aiida.calculations.arithmetic.add import ArithmeticAddCalculation
+from aiida.engine.processes.builder import ProcessBuilder
+from aiida import orm
+
+
+#@pytest.mark.usefixtures('clear_database_before_test')
+def test_access_methods():
+    """Test for the access methods (setter, getter, delete)."""
+    builder = ProcessBuilder(ArithmeticAddCalculation)
+
+    node_numb = orm.Int(4)
+    node_dict = orm.Dict(dict={'value': 4})
+
+    # AS ITEMS
+    builder['x'] = node_numb
+    assert dict(builder) == {'metadata': {'options': {}}, 'x': node_numb}
+
+    del builder['x']
+    assert dict(builder) == {'metadata': {'options': {}}}
+
+    with pytest.raises(ValueError):
+        builder['x'] = node_dict
+
+    builder['x'] = node_numb
+    assert dict(builder) == {'metadata': {'options': {}}, 'x': node_numb}
```

Comment:
This is indeed what I meant with my comment that you weren't sure you understood. Would be good to add a quick comment on why we are doing this, because it is a literal repeat of the first few lines. Something like:
```
# Set the same node again to check the setter still works despite deleting the item/attribute
```